### PR TITLE
[Feature]: Heatmap & AreaGraph Legends

### PR DESCRIPTION
**Pull request by nxanil**
@killanch  @chaitanyahalbe 

Fixed issue #434. Please review this PR. 

heatmap

### CODE REVIEW

**Comment by myTechPlayArea** [comment]
@nxanil Is issue 1) which is displaying all available heatmapColor object fixed? I dont see that in the gifs added. Can you please clarify.

**Comment by nxanil** [comment]
> @nxanil Is issue 1) which is displaying all available heatmapColor object fixed? I don't see that in the gifs added. Can you please clarify.

@myTechPlayArea 
That issue is also fixed in this PR. Please check the below mention screenshot.

`"heatmapColor": {
"0.0% - 0.49%": "#75B35E",
"0.5% - 0.99%": "#A8B343",
"1.0% - 1.49%": "#FAC956",
"1.5% - 1.99%": "#FFA939",
"2.0% - 3.99%": "#FF8550",
"4.0% - 5.99%": "#FF4949",
"6.0% - 9.99%": "#FF0D0D",
">= 10.0%": "#F2F3F4"
}`

heat

**Comment by chaitanyahalbe** [comment]
> > @nxanil Is issue 1) which is displaying all available heatmapColor object fixed? I don't see that in the gifs added. Can you please clarify.
> 
> @myTechPlayArea
> That issue is also fixed in this PR. Please check the below mention screenshot.
> 
> `"heatmapColor": { "0.0% - 0.49%": "#75B35E", "0.5% - 0.99%": "#A8B343", "1.0% - 1.49%": "#FAC956", "1.5% - 1.99%": "#FFA939", "2.0% - 3.99%": "#FF8550", "4.0% - 5.99%": "#FF4949", "6.0% - 9.99%": "#FF0D0D", ">= 10.0%": "#F2F3F4" }`
> 
> heat

@nxanil The Area Graph legend looks fine. But the legend for heatmap should appear in the same order as mentioned in the config file to avoid confusion.

**Comment by nxanil** [comment]
> > > @nxanil Is issue 1) which is displaying all available heatmapColor object fixed? I don't see that in the gifs added. Can you please clarify.
> > 
> > 
> > @myTechPlayArea
> > That issue is also fixed in this PR. Please check the below mention screenshot.
> > `"heatmapColor": { "0.0% - 0.49%": "#75B35E", "0.5% - 0.99%": "#A8B343", "1.0% - 1.49%": "#FAC956", "1.5% - 1.99%": "#FFA939", "2.0% - 3.99%": "#FF8550", "4.0% - 5.99%": "#FF4949", "6.0% - 9.99%": "#FF0D0D", ">= 10.0%": "#F2F3F4" }`
> > heat
> 
> @nxanil The Area Graph legend looks fine. But the legend for heatmap should appear in the same order as mentioned in the config file to avoid confusion.

@chaitanyahalbe 

Changes have been done. Please have a look.

heatmap

**Comment by chaitanyahalbe** [comment]
> > > > @nxanil Is issue 1) which is displaying all available heatmapColor object fixed? I don't see that in the gifs added. Can you please clarify.
> > > 
> > > 
> > > @myTechPlayArea
> > > That issue is also fixed in this PR. Please check the below mention screenshot.
> > > `"heatmapColor": { "0.0% - 0.49%": "#75B35E", "0.5% - 0.99%": "#A8B343", "1.0% - 1.49%": "#FAC956", "1.5% - 1.99%": "#FFA939", "2.0% - 3.99%": "#FF8550", "4.0% - 5.99%": "#FF4949", "6.0% - 9.99%": "#FF0D0D", ">= 10.0%": "#F2F3F4" }`
> > > heat
> > 
> > 
> > @nxanil The Area Graph legend looks fine. But the legend for heatmap should appear in the same order as mentioned in the config file to avoid confusion.
> 
> @chaitanyahalbe
> 
> Changes have been done. Please have a look.
> 
> heatmap

@nxanil Looks good.

**Review by chaitanyahalbe — APPROVED** [review]
Looks good.

File: Graphs/AbstractGraph.js [hunk 1]
```diff
@@ -547,19 +547,29 @@ export default class AbstractGraph extends React.Component {
         return legend.circleSize * circleToPixel;
     }
 
-    renderLegend(data, legend, getColor, label, isVertical) {
+    renderLegend(data, legend, getColor, label, isVertical, heatmapColor) {
```
Comment (killanch) [inline Graphs/AbstractGraph.js:550]: heatmapColor = {}
pls fix other such code as well, null heatmapColor should not result in exception

Comment (nxanil) [reply Graphs/AbstractGraph.js]: @killanch 

Changes have been made. Please check once.

**Review by killanch — APPROVED** [review]

**Comment by killanch** [comment]
@chaitanyahalbe can you pls merge and bring in to vsd-react-ui 0.0 and 20.10?

**Pull request merged**